File: backend/controllers/bookingController.js
const { v4: uuidv4 } = require('uuid');

const bookClass = async (req, res) => {
  const { userId, classId } = req.body;
  const booking = {
    id: uuidv4(),
    userId,
    classId,
    status: 'confirmed',
    createdAt: new Date().toISOString()
  };

  // Here: call Lambda to process payment

  res.status(201).json({ message: 'Booking successful', booking });
};

module.exports = { bookClass };

File: backend/controllers/booking_controller.go
package controllers

import (
	"encoding/json"
	"net/http"
	"time"

	"github.com/google/uuid"
)

type bookingRequest struct {
	UserID  string `json:"userId"`
	ClassID string `json:"classId"`
}

func BookClass(w http.ResponseWriter, r *http.Request) {
	var req bookingRequest
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		writeJSON(w, http.StatusBadRequest, map[string]string{"error": "invalid request body"})
		return
	}

	booking := map[string]any{
		"id":        uuid.NewString(),
		"userId":    req.UserID,
		"classId":   req.ClassID,
		"status":    "confirmed",
		"createdAt": time.Now().UTC().Format(time.RFC3339),
	}

	writeJSON(w, http.StatusCreated, map[string]any{
		"message": "Booking successful",
		"booking": booking,
	})
}
